.github/workflows/lint.yml

```
@@ -0,0 +1,24 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

Same concern as I had [here](https://github.com/hetznercloud/hcloud-python/pull/214#discussion_r1239641345)

```suggestion
  pull_request:
```